### PR TITLE
delete tree if delete node is initiated for an empty tree

### PR DESCRIPTION
**Pull request by philippotto**
### Mailable description of changes:
 - When deleting the last node of a tree, that tree will not be removed automatically anymore. Instead, the tree will just be empty. To remove that active tree, the "delete" shortcut can be used again.

### URL of deployed dev instance (used for testing):
- https://deleteemptytreewithdel.webknossos.xyz/

### Steps to test:
- create a new tree and delete it with del again
- test normal delete node functionality

### Issues:
- https://discuss.webknossos.org/t/removing-last-node-of-a-tree-also-deletes-tree-but-pressing-del-in-an-empty-tree-doesnt-do-anything/963

------
- [X] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
@daniel-wer Should be ready for final review :)

File: app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js [hunk 1]
```diff
@@ -201,14 +214,23 @@ test("SkeletonTracing should delete several nodes from a tree", t => {
...
-  t.not(newStateA, newState);
-  t.is(newStateB, newStateA);
+  const newState = ChainReducer(emptyTreeState)
+    .apply(SkeletonTracingReducer, createNodeAction)
+    .apply(SkeletonTracingReducer, createNodeAction)
+    .apply(SkeletonTracingReducer, deleteNodeAction)
+    .apply(SkeletonTracingReducer, deleteNodeAction)
+    .unpack();
+
+  t.deepEqual(
```
Comment (daniel-wer) [inline app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js]: Shouldn't `t.deepEqual(emptyTreeState, newState)` be true here?

File: app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js [hunk 2]
```diff
@@ -394,3 +369,36 @@ export const setTreeGroupAction = (groupId: ?string, treeId: number): SetTreeGro
   groupId,
   treeId,
 });
+
+// The following actions have the prefix "AsUser" which means that they
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js]: Nice, do you think moving the `deleteTreeWithConfirmAction` here as well would be sensible?

File: app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js (hunk 1 above)
Comment (philippotto) [reply app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js]: Unfortunately not :( The DiffableMap node collection creates an internal chunk when nodes are added, which is not removed again (it's just emptied). That's why the states differ. Ideally, the equal method would not compare the internal state of the DiffableMap, but I don't know how we can achieve this consistently without doing magic in an own compare-function..

File: app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js (hunk 2 above)
Comment (philippotto) [reply app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js]: Yes, good point, I'll do that!

File: app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js (hunk 1 above)
Comment (daniel-wer) [reply app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js]: I see, no big deal.

**Pull request merged**